packages/react/src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx

```diff
@@ -543,7 +543,10 @@ const TableBodyRow = ({
           data-nesting-offset={nestingOffset}
           onExpand={(evt) => stopPropagationAndCallback(evt, onRowExpanded, id, false)}
           onClick={() => {
-            if (shouldExpandOnRowClick) {
+            if (
```

Comment:
Nice, this should work. 
We have plenty of code duplication for the 2 `onClick` functions now so I would refactor and reference a single function from the JSX code. The only change to the function body would be the  `onRowSelected(id, !isExpanded)`;

```
const onRowClick = () => {
    if (shouldExpandOnRowClick && ((hasRowNesting && nestingChildCount) || hasRowExpansion)) {
      onRowExpanded(id, !isExpanded);
    }
    
    if (hasRowSelection === 'single' && isSelectable !== false) {
      onRowSelected(id, true);
    }
    if (isSelectable !== false) {
      onRowClicked(id);
    }
}
```